cmake/profile.cmake

```diff
@@ -1,4 +1,22 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-include(${MBED_ROOT}/cmake/profiles/${MBED_PROFILE}.cmake)
+# Mapping CMAKE_BUILD_TYPE into MBED_BUILD_TYPES, as we understand only 3 profiles
+set(MBED_BUILD_TYPES Debug Release Develop)
+
+# Force the build types to be case-insensitive for checking
+string(TOLOWER ${CMAKE_BUILD_TYPE} LOWERCASE_CMAKE_BUILD_TYPE)
+string(TOLOWER ${MBED_BUILD_TYPES} LOWERCASE_MBED_BUILD_TYPES)
+
+get_property(multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(multi_config)
+  set(CMAKE_CONFIGURATION_TYPES "${MBED_BUILD_TYPES}" CACHE STRING "List of supported build types" FORCE)
+else()
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Build type of the project.")
```

Comment:
Does this line mean that the default build type is `Release`? If so, should it not be `Develop` as with Mbed CLI?

Reply:
dont review until updated